addons/mrp/models/mrp_workcenter.py

```diff
@@ -195,12 +234,40 @@ def unblock(self):
     def create(self, vals_list):
         # resource_type is 'human' by default. As we are not living in
         # /r/latestagecapitalism, workcenters are 'material'
-        records = super(MrpWorkcenter, self.with_context(default_resource_type='material')).create(vals_list)
-        return records
+
+        resources_vals_list = []
+        calendar_ids = [vals['resource_calendar_id'] for vals in vals_list if vals.get('resource_calendar_id')]
+        calendars_tz = {calendar.id: calendar.tz for calendar in self.env['resource.calendar'].browse(calendar_ids)}
+        for vals in vals_list:
+            if not vals.get('resource_ids'):
+                workorder_simultaneous = vals.get('simultaneous_workorder', 1)
+                for _i in range(workorder_simultaneous):
+                    resources_vals_list.append(self._prepare_resource_values(
+                        vals,
+                        vals.pop('tz', False) or calendars_tz.get(vals.get('resource_calendar_id'))
+                    ))
+        if resources_vals_list:
+            resources = self.env['resource.resource'].create(resources_vals_list)
+            for vals, res in zip(vals_list, resources):
+                if not vals.get('resource_ids'):
+                    vals['resource_ids'] = res
+        return super(MrpWorkcenter, self.with_context(check_idempotence=True, default_resource_type='material')).create(vals_list)
```

Comment:
```suggestion
        return super().create(vals_list)
```
Set the material during ressource creation